app/controllers/carts_controller.rb

```diff
@@ -4,6 +4,8 @@ class CartsController < ApplicationController
   before_action :check_quantity_exceeds_amount, only: :update
   skip_before_action :verify_authenticity_token, only: :update
 
+  authorize_resource class: CartsController
```

Comment:
thằng ni để làm chi em?
```suggestion
  authorize_resource
```

Reply:
cái authorize_resource sẽ auto checjk phân quyền cho mỗi action trong controller.
cái khai báo class: em thêm vào vì những controller không có model sẽ báo lỗi không tìm được model và  không chạy. Khi em thêm vào khai báo class: thì phân quyền đó chạy bình thường.